src/components/implementation/capmgr/naive/init.c

```diff
@@ -72,6 +73,26 @@ capmgr_comp_info_iter(void)
 		} else if (cos_spd_id() == spdid) {
 			cap_info_initthd_init(rci, sl__globals()->sched_thd, 0);
 		}
+
+		/*
+		 * Set tls region for this component
+		 * Expand the 2nd level pte within this component's page table at our new range
+		 */
+		tmp.pgtbl_cap = pgtslot;
+		tmp.memsrc    = ci;
+		ret = (int)cos_pgtbl_intern_alloc(&tmp, pgtslot, TLS_BASE_ADDR,
+					     PAGE_SIZE * TLS_NUM_PAGES);
+		assert(ret);
+
+		/* Place desired number of pages at this new range */
+		offset = 0;
+		for (i = 0 ; i < (signed)TLS_NUM_PAGES ; i++) {
+			src_pg = (vaddr_t)cos_page_bump_alloc(ci);
+			assert(src_pg);
+			ret = cos_mem_alias_at(&tmp, TLS_BASE_ADDR + offset, ci, src_pg);
+			assert(!ret);
+			offset += PAGE_SIZE;
+		}
```

Comment:
I'd have all of this in `cap_info.h/.c` API like cap_info_comptls_init() or something and call that! There is so much code already in this function, I'd avoid expanding it when there are other alternatives.. 

Better, I'd call that `cap_info_comptls_init()` in `cap_info_comp_init` function..

Reply:
Thanks for the advice on this, being unfamiliar with a lot of the changes to the llbooter, I had no idea where it would be ideal to place this initialization code. I selected this location as it worked haha